test/duplicate_flag_name.cc

```diff
@@ -12,3 +14,110 @@ TEST(FlagsTest, DuplicateFlagName) {
       "Encountered duplicate flag name 'same' for "
       "field 'test.DuplicateFlagName.s2'");
 }
+
+TEST(FlagsTest, DuplicateSucceed) {
+  test::Flags flags;
+
+  auto parser = stout::flags::Parser::Builder(&flags).Build();
+
+  std::array arguments = {
+      "/path/to/program",
+      "--foo='hello world'",
+      "--bar=true",
+      "--bar=true",
+      "--bar",
+      "--bar",
+  };
+
+  int argc = arguments.size();
+  const char** argv = arguments.data();
+
+  parser.Parse(&argc, &argv);
+
+  EXPECT_TRUE(flags.bar());
+}
+
+TEST(FlagsTest, DuplicateConflict) {
+  test::DuplicateFlagsDeath duplicates;
+
+  auto parser = stout::flags::Parser::Builder(&duplicates).Build();
+
+  std::array arguments = {
+      "/path/to/program",
+      "--duplicate",
+      "--duplicate=false",
+      "--no-b",
+      "--b",
+  };
+
+  int argc = arguments.size();
+  const char** argv = arguments.data();
+
+// On POSIX systems (e.g. Linux, Cygwin, and Mac), googletest
+// uses the POSIX extended regular expression syntax. To learn
+// about this syntax, you may want to check the link below:
+// https://tinyurl.com/yta5f24r
+// On Windows, googletest uses its own simple regular
+// expression implementation. Check also the link below.
+// https://tinyurl.com/2r59uuuv
+#ifdef _WIN32
+  const std::string regex =
+      "program: Failed while parsing "
+      "and validating flags:"
+      "\\n\\n"
+      "\\* Encountered duplicate boolean flag 'b' "
+      "with flag aliased as 'no-b' "
+      "that has a conflicting value"
+      "\\n\\n"
+      "\\* Encountered duplicate boolean flag 'duplicate' "
+      "that has a conflicting value";
+#else
+  const std::string regex =
+      "program: Failed while parsing "
+      "and validating flags:"
+      ".+"
+      "Encountered duplicate boolean flag 'b' "
+      "with flag aliased as 'no-b' "
+      "that has a conflicting value"
+      ".+"
+      "Encountered duplicate boolean flag 'duplicate' "
+      "that has a conflicting value";
+#endif
```

Comment:
See my comment in your other review about just always using the simple regular expressions to avoid needing to duplicate them with `#ifdef`.